### PR TITLE
booking and ID complete

### PR DESCRIPTION
**Pull request by mmmlaphroaig**

### CODE REVIEW

File: src/main/java/com/keysoft/mongodb/controller/QacController.java [hunk 1]
```diff
@@ -203,5 +224,63 @@ public String cleanComment(String originalComment) {
         }
         return profanityWords;
     }
+
+
+    public String GetTimeAndID(String a) {
```
Comment (tvaidotas) [inline src/main/java/com/keysoft/mongodb/controller/QacController.java]: what's a meant to be? not a good name

File: src/main/java/com/keysoft/mongodb/controller/QacController.java [hunk 2]
```diff
@@ -203,5 +224,63 @@ public String cleanComment(String originalComment) {
...
+    	DateTimeFormatter time = DateTimeFormatter.ofPattern("dd/MM/YYYY HH:mm:ss");
+    	DateTimeFormatter id = DateTimeFormatter.ofPattern("ssddHHmmMM");
+    	LocalDateTime now = LocalDateTime.now();
+    	String b = "";
```
Comment (tvaidotas) [inline src/main/java/com/keysoft/mongodb/controller/QacController.java]: what's b meant to be?

File: src/main/java/com/keysoft/mongodb/controller/QacController.java [hunk 3]
```diff
@@ -203,5 +224,63 @@ public String cleanComment(String originalComment) {
...
+    	}  
+    	return b;
+    }
+    
+    public static String InitialGrabber(String firstName, String secondName) {
+        String fName = firstName;
+        String sName = secondName;
+        String initials = "";
+        Pattern pattern = Pattern.compile("(?<=^[A-Z]).*");  
+        String[] firstInitial = pattern.split(fName);
```
Comment (tvaidotas) [inline src/main/java/com/keysoft/mongodb/controller/QacController.java]: why do you have fName variable if you use it once, just put the firstName directly into split
same with sName

File: src/main/java/com/keysoft/mongodb/controller/QacController.java [hunk 4]
```diff
@@ -203,5 +224,63 @@ public String cleanComment(String originalComment) {
...
+        String[] secondInitial = pattern.split(sName);
+        for (String temp: firstInitial) 
+        	initials += temp;
+        for (String temp: secondInitial)
+        	initials += temp;
+        return initials;
+//        TADAS' LOUSY EFFORT:
+//        System.out.println(Character.toString(firstName.charAt(0)) + Character.toString(secondName.charAt(0)));
```
Comment (tvaidotas) [inline src/main/java/com/keysoft/mongodb/controller/QacController.java]: you took a one line statement that worked and made it into this mess of over complicating

File: src/main/java/com/keysoft/mongodb/controller/QacController.java [hunk 5]
```diff
@@ -203,5 +224,63 @@ public String cleanComment(String originalComment) {
...
+        
+    private static String generateQR(String text, int width, int height) throws WriterException, IOException {
+    	QRCodeWriter qrCodeWriter = new QRCodeWriter();
+    	BitMatrix bitMatrix = qrCodeWriter.encode(text, BarcodeFormat.QR_CODE, width, height);
+    	
+    	ByteArrayOutputStream pngOutputStream = new ByteArrayOutputStream();
+    	MatrixToImageWriter.writeToStream(bitMatrix, "PNG", pngOutputStream);
+    	byte[] pngData = pngOutputStream.toByteArray();
+    	String base = Base64.getEncoder().encodeToString(pngData);
+    	//	System.out.println("data:image/png;base64,"+base);
```
Comment (tvaidotas) [inline src/main/java/com/keysoft/mongodb/controller/QacController.java]: is this actually needed?

**Comment by mmmlaphroaig** [comment]
cheers Tadas, new one ULd

**Pull request merged**